includes/sanitizers/class-amp-gallery-block-sanitizer.php

```diff
@@ -78,35 +78,30 @@ public function sanitize() {
 				continue;
 			}
 
-			$num_images = 0;
+			$images = null;
 
 			// If it's not AMP lightbox, look for links first.
 			if ( ! $is_amp_lightbox ) {
-				$images     = $node->getElementsByTagName( 'a' );
-				$num_images = $images->length;
+				$images = $node->getElementsByTagName( 'a' );
 			}
 
-			if ( 0 === $num_images ) {
-
-				// If not linking to anything then look for <amp-img>.
-				$images     = $node->getElementsByTagName( 'amp-img' );
-				$num_images = $images->length;
+			// If not linking to anything then look for <amp-img>.
+			if ( ! $images || 0 === $images->length ) {
+				$images = $node->getElementsByTagName( 'amp-img' );
 			}
 
-			if ( 0 === $num_images ) {
+			// Skip if no images found.
+			if ( ! $images || 0 === $images->length ) {
 				continue;
 			}
 
-			$carousel_height     = $this->get_carousel_height( $node );
-			$carousel_attributes = array(
-				'height' => $carousel_height,
+			$amp_carousel = AMP_DOM_Utils::create_node( $this->dom, 'amp-carousel', array(
+				'height' => $this->get_carousel_height( $node ),
 				'type'   => 'slides',
 				'layout' => 'fixed-height',
-			);
-			$amp_carousel        = AMP_DOM_Utils::create_node( $this->dom, 'amp-carousel', $carousel_attributes );
-
-			for ( $j = $num_images - 1; $j >= 0; $j-- ) {
-				$amp_carousel->appendChild( $images->item( $j ) );
+			) );
+			foreach ( $images as $image ) {
+				$amp_carousel->appendChild( $image );
```

Comment:
@westonruter `for ( $j = $num_images - 1; $j >= 0; $j-- ) {` was used here since when looping directly through the nodes (`foreach ( $images as $image ) {`) it's not working as expected, for example when adding a gallery with 5 images locally and using AMP Carousel it will just display the first three in the gallery. That's happening when testing locally now as well, is it working as expected for you?

Reply:
PR to fix opened in #1187.